### PR TITLE
Fix deprecated errors of dependency injection

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Fix deprecated errors of dependency injection [using a factory to create services](http://symfony.com/doc/current/components/dependency_injection/factories.html)

### CODE REVIEW

**Comment by bocharsky-bw** [comment]
Hi @stof , Could you merge this doc fix?

**Comment by stof** [comment]
Can you add a note saying that this documentation uses a Symfony 2.6+ syntax and that people using older versions should check the Symfony documentation ? This will avoid getting reports from them saying our doc is wrong.

**Comment by bocharsky-bw** [comment]
@stof I added. Does it seems good for you?

**Comment by bocharsky-bw** [comment]
@stof ping

File: Resources/doc/menu_service.rst [hunk 1]
```diff
@@ -59,19 +59,23 @@ object created by the ``createMainMenu`` method:
...
-            factory_service: app.menu_builder
-            factory_method: createMainMenu
+            factory: ["@app.menu_builder", createMainMenu]
             arguments: ["@request_stack"]
             tags:
                 - { name: knp_menu.menu, alias: main } # The alias is what is used to retrieve the menu
 
         # ...
 
+>
```
Comment (stof) [inline Resources/doc/menu_service.rst]: this should be reverted

File: Resources/doc/menu_service.rst [hunk 2]
```diff
@@ -59,19 +59,23 @@ object created by the ``createMainMenu`` method:
...
         # ...
 
+> 
+
 .. note::
 
     The menu service must be public as it will be retrieved at runtime to keep
     it lazy-loaded.
 
+    If you are using Symfony `2.5` or older version please check the `Using a Factory to Create Services`_
```
Comment (stof) [inline Resources/doc/menu_service.rst]: this should be a second note, not something in the same note

Comment (bocharsky-bw) [inline Resources/doc/menu_service.rst]: OK

**Comment by stof** [comment]
thanks

**Pull request merged**